packages/angular_devkit/build_angular/src/tools/esbuild/bundler-context.ts

```diff
@@ -217,8 +231,27 @@ export class BundlerContext {
       }
     }
 
+    const outputFiles = result.outputFiles.map(({ text, path }) => {
+      let fileType: BuildOutputFileType;
+      if (dirname(path) === 'media') {
+        fileType = BuildOutputFileType.Media;
+      } else {
+        fileType =
+          this.#esbuildOptions?.platform === 'node'
+            ? BuildOutputFileType.Server
+            : BuildOutputFileType.Browser;
+      }
+
+      return createOutputFileFromText(path, text, fileType);
```

Comment:
The output files from esbuild have the `text` field lazily generated from the UInt8Array `contents` field.
Can you switch this to use the data helper function instead to avoid multiple conversions from/to UTF8?